### PR TITLE
[Part0] Support Activity Status and status description in Console Exporter.

### PR DESCRIPTION
**Pull request by Yun-Ting**
Related to https://github.com/open-telemetry/opentelemetry-dotnet/issues/2569.

## Changes
Added support to to set status with Activity.Status, which was introduced in `System.Diagnostic.DiagnosticSource 6.0.0` to ConsoleExporter.
The same idea will be applied to other exporters.

**Case 1:** 
Setting activity status to be OK.
`activity.SetStatus(ActivityStatusCode.Ok);`
Output:
`StatusCode : OK`

**Case 2:** 
Setting activity status to be ERROR with description.
`activity.SetStatus(ActivityStatusCode.Error, "desc");`
Output:
```
StatusCode : ERROR
Error : desc
```

**Case 3:** 
Setting activity tag to be OK.
`activity.SetTag("otel.status_code", "OK");`
Output:
```
Activity.Tags:
   StatusCode : OK
```

**Case 4:** 
Setting activity tag to be ERROR with description.
```
activity.SetTag("otel.status_code", "ERROR");
activity.SetTag("otel.status_description", "desc");
```
Output:
```
Activity.Tags:
   StatusCode : ERROR
   Error : desc
```

**Case 5:** 
Activity status take precedence over activity tag when both were set.
```
activity.SetStatus(ActivityStatusCode.Ok);
activity.SetTag("otel.status_code", "ERROR");
activity.SetTag("otel.status_description", "desc");
```
Output:
```
Activity.Tags:
StatusCode : OK
```
**Case 6:** 
Activity status take precedence over activity tag when both were set.
```
activity.SetStatus(ActivityStatusCode.Error, "desc");
activity.SetTag("otel.status_code", "OK");
```
Output:
```
Activity.Tags:
StatusCode : ERROR
Error : desc
```

For significant contributions please make sure you have completed the following items:

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Exporter.Console/ConsoleActivityExporter.cs [hunk 1]
```diff
@@ -71,25 +72,30 @@ public override ExportResult Export(in Batch<Activity> batch)
...
+                }
 
-                    if (!string.IsNullOrEmpty(statusCode))
+                if (activity.Status != ActivityStatusCode.Unset)
+                {
+                    this.WriteLine($"Activity statusCode : {StatusHelper.GetTagValueForActivityStatusCode(activity.Status)}");
+                    if (activity.Status == ActivityStatusCode.Error)
                     {
-                        this.WriteLine($"   StatusCode : {statusCode}");
+                        this.WriteLine($"error : {activity.StatusDescription ?? string.Empty}");
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.Console/ConsoleActivityExporter.cs:90]: minor: it seems the upper/lower case is a bit inconsistent in the Console Exporter (e.g. `StatusCode` is CamelCase).

**Review by reyang — APPROVED** [review]

File: src/OpenTelemetry.Exporter.Console/ConsoleActivityExporter.cs [hunk 2]
```diff
@@ -71,25 +72,30 @@ public override ExportResult Export(in Batch<Activity> batch)
...
 
                         this.WriteLine($"    {tag.Key}: [{string.Join(", ", array.Cast<object>())}]");
                     }
+                }
 
-                    if (!string.IsNullOrEmpty(statusCode))
+                if (activity.Status != ActivityStatusCode.Unset)
+                {
+                    this.WriteLine($"Activity StatusCode : {StatusHelper.GetTagValueForActivityStatusCode(activity.Status)}");
+                    if (activity.Status == ActivityStatusCode.Error)
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.Console/ConsoleActivityExporter.cs:88]: We don't need this check. This exporter can simply display the Description if not null.

File: src/OpenTelemetry.Exporter.Console/ConsoleActivityExporter.cs [hunk 3]
```diff
@@ -71,25 +72,30 @@ public override ExportResult Export(in Batch<Activity> batch)
...
                     {
-                        this.WriteLine($"   StatusCode : {statusCode}");
+                        this.WriteLine($"Error : {activity.StatusDescription ?? string.Empty}");
                     }
-
+                }
+                else if (!string.IsNullOrEmpty(statusCode))
+                {
+                    this.WriteLine($"   StatusCode : {statusCode}");
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.Console/ConsoleActivityExporter.cs:95]: please make sure the messages are consistent. "Activity StatusCode" vs "StatusCode".

File: src/OpenTelemetry.Exporter.Console/ConsoleActivityExporter.cs [hunk 4]
```diff
@@ -71,25 +72,30 @@ public override ExportResult Export(in Batch<Activity> batch)
...
                         }
 
                         this.WriteLine($"    {tag.Key}: [{string.Join(", ", array.Cast<object>())}]");
                     }
+                }
 
-                    if (!string.IsNullOrEmpty(statusCode))
+                if (activity.Status != ActivityStatusCode.Unset)
+                {
+                    this.WriteLine($"Activity StatusCode : {StatusHelper.GetTagValueForActivityStatusCode(activity.Status)}");
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.Console/ConsoleActivityExporter.cs:87]: I think we can simply do 
`this.WriteLine($"   StatusCode : {activity.Status}");`

No need of the helper methods, for this console exporter

File: src/OpenTelemetry.Exporter.Console/ConsoleActivityExporter.cs (hunk 3 above)
Comment (Yun-Ting) [reply src/OpenTelemetry.Exporter.Console/ConsoleActivityExporter.cs:94]: Updated to be "StatusCode".

File: src/OpenTelemetry.Exporter.Console/ConsoleActivityExporter.cs [hunk 5]
```diff
@@ -71,25 +71,30 @@ public override ExportResult Export(in Batch<Activity> batch)
...
+                }
 
-                    if (!string.IsNullOrEmpty(statusCode))
+                if (activity.Status != ActivityStatusCode.Unset)
+                {
+                    this.WriteLine($"StatusCode : {activity.Status}");
+                    if (!string.IsNullOrEmpty(activity.StatusDescription))
                     {
-                        this.WriteLine($"   StatusCode : {statusCode}");
+                        this.WriteLine($"Error : {activity.StatusDescription}");
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.Console/ConsoleActivityExporter.cs:89]: Remove "Error". Just Status Description is good enough here.

**Review by cijothomas — APPROVED** [review]
nit comment. may be addressed separate

File: src/OpenTelemetry.Exporter.Console/ConsoleActivityExporter.cs (hunk 5 above)
Comment (Yun-Ting) [reply src/OpenTelemetry.Exporter.Console/ConsoleActivityExporter.cs:89]: Maybe something like:
`this.WriteLine($"Status Description : {activity.StatusDescription}");`
?

Comment (cijothomas) [reply src/OpenTelemetry.Exporter.Console/ConsoleActivityExporter.cs:89]: yes

Comment (Yun-Ting) [reply src/OpenTelemetry.Exporter.Console/ConsoleActivityExporter.cs:89]: https://github.com/open-telemetry/opentelemetry-dotnet/pull/3064

**Pull request merged**